core/src/main/java/org/mule/runtime/core/internal/connection/PoolingConnectionProviderWrapper.java

```diff
@@ -32,9 +34,11 @@
    */
   public PoolingConnectionProviderWrapper(ConnectionProvider<C> delegate,
```

Comment:
What about having a wrapper that know how to resolve the config name, and have all the connection provider wrapped with it. 

I think that this is not exclusive to PoolingConnectionProviders, and that in the future we may need it for the others.

Reply:
What is the advantage of creating a new wrapper? If others want to add this functionality they just override the method and provide it.

Reply:
We would avoid the need of other wrappers to be in charge of getting the owner config name.

Reply:
re-review now that the default implementation was done at the interface

Reply:
What happens now is that the only instance of ConnectionProviderWrapper that can resolve the owner config name, are the ones that contain a PoolingWrapper. Wrapping all CPW in one that resolves the name will make it available for all CPs.

Reply:
Sure, if we want the name available for all CPs then I agree to make a new wrapper.